app/src/main/java/com/courseproject/tindar/BlankNavActivity.java

```diff
@@ -17,31 +18,32 @@
 import com.courseproject.tindar.databinding.ActivityBlankNavBinding;
 
 public class BlankNavActivity extends AppCompatActivity {
-
     private AppBarConfiguration mAppBarConfiguration;
-    private ActivityBlankNavBinding binding;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        binding = ActivityBlankNavBinding.inflate(getLayoutInflater());
+        // retrieves user Id passed from other activity
+        Intent intent = getIntent();
+        String userId = intent.getStringExtra("user_id");
+
+        // instantiates the blank nav view model and sets the user id that is currently logged in
+        BlankNavViewModel blankNavViewModel = new ViewModelProvider(this).get(BlankNavViewModel.class);
```

Comment:
Do we have to make a new one here, or is it possible to get it passed in from the previous screen?

Reply:
This is a view model for Blank Nav activity - so it shares the data between the activity and its fragments. I saw this in google, so I think we cannot share the view model across the activities.

<img width="747" alt="image" src="https://github.com/CSC207-2023Y-UofT/course-project-tindar/assets/76457965/ec9823e1-ebd0-4b04-ae1e-9903193ebb8f">